### DISCUSSION TITLE
How to generate fireware about batch_size=n?

### DISCUSSION THREAD

**Discussion by zyt1024** [fastmachinelearning/hls4ml#952] [Q&A]
If I want to handle data with` batch_size=2`, can hls4ml generate the corresponding HLS code?

**Comment by vloncar** [comment]
No. You can manually modify the generated HLS to loop over the samples, but that may not be the most efficient implementation.

**Reply by manuelblancovalentin** [reply]
Opening this again, and just to clarify: during inference batch size is assumed to be always 1, right?

**Reply by vloncar** [reply]
Always 1
